src/Databases/Iceberg/RestCatalog.cpp

```diff
@@ -0,0 +1,622 @@
+#include <Databases/Iceberg/RestCatalog.h>
+#include <Databases/Iceberg/StorageCredentials.h>
+
+#include <base/find_symbols.h>
+#include <Core/Settings.h>
+#include <Common/escapeForFileName.h>
+#include <Common/threadPoolCallbackRunner.h>
+#include <Common/Base64.h>
+
+#include <IO/ConnectionTimeouts.h>
+#include <IO/HTTPCommon.h>
+#include <IO/ReadBuffer.h>
+#include <Interpreters/Context.h>
+
+#include <Storages/ObjectStorage/DataLakes/IcebergMetadata.h>
+#include <Server/HTTP/HTMLForm.h>
+#include <Formats/FormatFactory.h>
+
+#include <Poco/URI.h>
+#include <Poco/JSON/Array.h>
+#include <Poco/JSON/Parser.h>
+
+namespace DB::ErrorCodes
+{
+    extern const int ICEBERG_CATALOG_ERROR;
+    extern const int LOGICAL_ERROR;
+    extern const int BAD_ARGUMENTS;
+    extern const int NOT_IMPLEMENTED;
+}
+
+namespace DB::Setting
+{
+    extern const SettingsBool iceberg_engine_ignore_schema_evolution;
+}
+
+namespace Iceberg
+{
+
+static constexpr auto config_endpoint = "config";
+static constexpr auto namespaces_endpoint = "namespaces";
+
+namespace
+{
+
+std::pair<std::string, std::string> parseCatalogCredential(const std::string & catalog_credential)
```

Comment:
Maybe reuse BasicAuth object https://github.com/ClickHouse/ClickHouse/blob/1d0caf1c906f359b90ed7c6a266417a463cfbe34/src/Access/Credentials.h#L56-L70?

Reply:
Or this pair is not basic auth?

Reply:
yes, that is not it